src/utils/api.ts

```diff
@@ -16,7 +18,11 @@ export const badRequest = (res: NextApiResponse) => {
   return res.status(400).end();
 };
 
-export const notSignedIn = (res: NextApiResponse) => {
+export const notSignedIn = (res: NextApiResponse, url?: string) => {
+  if (url && (url as string).endsWith('api/csv/issue')) {
+    return res.status(401).json({ provider: 'gitlab' });
+  }
+
```

Comment:
That shouldn't be here, but in the integration on `api/csv/issue`. Maybe instead of pass `url`as params, we should have a param to pass as `json` content directly.